### PR TITLE
Add hook to insert statsd client to measure amorphic internal stats

### PR DESCRIPTION
**Pull request by nsaenz7**
things that were changed:
- add a statsd client and place it onto supertype for use by other amorphic subcomponents
- implement timing functions in important areas of amorphic codebase

things that didn't change:
- testing infrastructure is not good. going to fix with more 'real world' updated typescript tests in a separate ticket.

### CODE REVIEW

**Review by sshankar19 — APPROVED** [review]

File: tsconfig.json [hunk 1]
```diff
@@ -1,5 +1,5 @@
 {
-  "include": [
+  "files": [
     "./lib/AmorphicServer.ts"
```
Comment (sshankar19) [inline tsconfig.json:3]: why'd you change this?

File: lib/routes/processMessage.js [hunk 2]
```diff
@@ -129,8 +144,14 @@ function processMessage(req, res, sessions, nonObjTemplatelogLevel, controllers)
                 }
 
                 ourObjectTemplate.logger.clearContextProps(context);
+
                 res.end(respstr);
```
Comment (sshankar19) [inline lib/routes/processMessage.js:147]: whitespace?

File: lib/routes/processMessage.js [hunk 3]
```diff
@@ -152,12 +173,22 @@ function processMessage(req, res, sessions, nonObjTemplatelogLevel, controllers)
...
+
+                statsdUtils.computeTimingAndSend(
+                    processMessageStartTime,
+                    'amorphic.webserver.process_message.response_time',
+                    { result: 'failure' });
             }
 
         }).catch(function failure(error) {
-            log(0, req.session.id, error.message + error.stack, nonObjTemplatelogLevel);
+            // log(0, req.session.id, error.message + error.stack, nonObjTemplatelogLevel);
```
Comment (sshankar19) [inline lib/routes/processMessage.js]: do we not want this log thing?

File: lib/session/compressSessionData.js [hunk 4]
```diff
@@ -1,21 +1,37 @@
...
+                compressSessionDataStartTime,
+                'amorphic.session.compress_session_data.response_time',
+                { result: 'success' });
+
+            return sessionData;
+        } catch (e) {
+            statsdUtils.computeTimingAndSend(
+                compressSessionDataStartTime,
+                'amorphic.session.compress_session_data.response_time',
+                { result: 'success' });
```
Comment (sshankar19) [inline lib/session/compressSessionData.js]: result: 'success'?

**Review by sshankar19 — CHANGES_REQUESTED** [review]

File: tsconfig.json (hunk 1 above)
Comment (nsaenz7) [reply tsconfig.json:3]: not a super significant change, they do different flavors of the same thing. `include` is used when you're using some sort of pattern matching, but we're explicitly saying "I want this file and only this file", so I used the tsconfig prop that seemed to best match that behavior, `files`.

File: lib/routes/processMessage.js (hunk 3 above)
Comment (nsaenz7) [reply lib/routes/processMessage.js]: nope! we do. will uncomment

File: lib/session/compressSessionData.js (hunk 4 above)
Comment (nsaenz7) [reply lib/session/compressSessionData.js]: darn, will change to `failure`. good catch!

File: lib/routes/processMessage.js (hunk 2 above)
Comment (nsaenz7) [reply lib/routes/processMessage.js:147]: this was intentional, I just wanted to see some more space here haha

**Review by sshankar19 — DISMISSED** [review]

**Review by sshankar19 — APPROVED** [review]

**Review by nikmash — APPROVED** [review]

**Pull request merged**